src/pages/FlagLabware.tsx

```diff
@@ -0,0 +1,183 @@
+import PinkButton from '../components/buttons/PinkButton';
+import OperationCompleteModal from '../components/modal/OperationCompleteModal';
+import React, { useCallback, useContext, useMemo } from 'react';
+import { StanCoreContext } from '../lib/sdk';
+import createFormMachine from '../lib/machines/form/formMachine';
+import { FlagDetail, FlagLabwareMutation, FlagLabwareRequest, LabwareFieldsFragment } from '../types/sdk';
+import { useMachine } from '@xstate/react';
+import * as Yup from 'yup';
+import Panel from '../components/Panel';
+import RemoveButton from '../components/buttons/RemoveButton';
+import DataTable from '../components/DataTable';
+import { FormikErrorMessage } from '../components/forms';
+import { LabwareFlagDetails } from '../components/labwareFlagDetails';
+
+type FormFlagLabware = {
+  labware: LabwareFieldsFragment | undefined;
+  description: string;
+};
+
+const initialValues: FormFlagLabware = { labware: undefined, description: '' };
+function buildValidationSchema() {
+  return Yup.object().shape({
+    labware: Yup.object().label('Labware').required('Labware is required'),
+    description: Yup.string()
+      .label('Description')
+      .required('Description is required')
+      .max(500, 'Please enter a description of no more than 500 characters')
+      .min(5, 'Please enter a valid description')
+  });
+}
+
+const FlagLabware = () => {
+  const stanCore = useContext(StanCoreContext);
+  const formMachine = React.useMemo(() => {
+    return createFormMachine<FlagLabwareRequest, FlagLabwareMutation>().withConfig({
+      services: {
+        submitForm: (ctx, e) => {
+          if (e.type !== 'SUBMIT_FORM') return Promise.reject();
+          return stanCore.FlagLabware({ request: e.values });
+        }
+      }
+    });
+  }, [stanCore]);
+  const [current, send] = useMachine(() => formMachine);
+  const validationSchema = useMemo(() => buildValidationSchema(), []);
+
+  const convertValuesAndSubmit = async (values: FormFlagLabware) => {
+    if (!values.labware) return;
+    const requestValues: FlagLabwareRequest = {
+      barcode: values.labware!.barcode,
+      description: values.description
+    };
+    send({ type: 'SUBMIT_FORM', values: requestValues });
+  };
+
+  const serverError = current.context.serverError;
+
+  const labwareColumns = useMemo(() => {
+    return [columns.barcode(), columns.externalName(), columns.donorId(), columns.tissueType(), columns.labwareType()];
+  }, []);
+
+  const [relatedFlags, setRelatedFlags] = React.useState<FlagDetail[]>([]);
+
+  const checkRelatedFlagges = useCallback(
+    async (labwares: LabwareFieldsFragment[], foundLabware: LabwareFieldsFragment): Promise<string[]> => {
+      return stanCore
+        .GetLabwareFlagDetails({
+          barcodes: [foundLabware.barcode]
+        })
+        .then((response) => {
+          setRelatedFlags(response.labwareFlagDetails);
+          return [];
+        });
```

Comment:
Using await will improve readability, also allows  try/catch blocks for error handling